### DISCUSSION TITLE
How can i properly start recording when component mount

### DISCUSSION THREAD

**Discussion by batmnkh9608** [katspaugh/wavesurfer.js#3309] [Q&A]
How can i properly start recording when component mount
Hi im new to React.js. Im having this problem and i hope i will get answer

Im conditionally rendering Child Component that uses record plugin
if condition true the component mounts and start recording automatically 

For Example : 
in Parent Component
`isRecording ? <ChildComponent /> : <AnotherComponent>`

in Child Component
`useEffect(() => {
        const ws = WaveSurfer.create({
          options
        })

    RecordPlugin.getAvailableAudioDevices().then((devices: any) =>
      setDeviceId(devices[0].deviceId)
    )

    const recordPlugin = ws.registerPlugin(RecordPlugin.create())
    setRecord(recordEffect)

    recordPlugin.on("record-end", (blob: any) => {
      sendAudio(blob)
    })

    recordPlugin.startRecording({deviceId})

    return () => {
      ws.destroy()
    }
  }, [])`
  
   ` const stopRecording = () => {
    record.stopRecording()
    record.stopMic()
    setIsRecording(false)
  }`

when i click stop recording in child component it set IsRecording to false and the component unmount
but the recording icon on tab still shows how to fix this problem

**Comment by katspaugh** [comment]
Try stopping the mic before the destroy. It might be a bug if it’s not done automatically.

**Reply by batmnkh9608** [reply]
i tried but still same
